app/script/ui/ViewportObserver.js

```diff
@@ -38,10 +38,12 @@ z.ui.ViewportObserver = (() => {
     const onIntersect = entries => {
       entries.forEach(entry => {
         if (entry.isIntersecting) {
-          const trackedElement = _getObservedElement(entry.target);
-          trackedElement.callback();
-          intersectionObserver.unobserve(trackedElement.element);
-          _removeObservedElement(trackedElement);
+          const observedElement = _getObservedElement(entry.target);
+          if (observedElement) {
+            observedElement.callback();
+            intersectionObserver.unobserve(observedElement.element);
```

Comment:
We might want to unobserve the `entry.target` still (even if there is not matching element)